### ISSUE TITLE
Procedural map generation fails for maps larger than 16x16km

### ISSUE THREAD

**Issue by Insprill** [Insprill/dv-mapify#58]
We need a different approach to generating the height map, since a single texture can't be more than 16x16k pixels.

**Comment by WhistleWiz** [comment]
This only affects the map item?
We could simply, on larger sizes, average pixels.
So instead of doing it 1m = 1px, we'd average a 2x2 square into a single pixel. Bigger even, 4x4, etc.

**Comment by Insprill** [comment]
Yeah, it's only items. It depends on the terrain resolution as well, since that's what determines the texture size.
https://github.com/Insprill/dv-mapify/blob/24dcdfd261ba14634dfd82e37e9d44b6e6543e92/MapifyEditor/Procedural/MapRenderer.cs#L24-L28
That's a pretty bad way of doing it, and we should instead pack it into a 4k texture, then smooth the value on small maps and down sample it on larger ones.

**Comment by t0stiman** [comment]
Maybe we could create a world map by placing an orthogonal projection camera above the world, and having that snap a 'picture'. Then the world map would automatically have the correct terrain colors.

**Comment by Insprill** [comment]
It's not really supposed to have the correct colors of the terrain, it's just a heightmap that we color by elevation, which is what the vanilla one is.

We could use a camera and just save a depth buffer, that would also get us what we need, and would likely be simpler than whatever other solution we'd come up with to continue using the current approach.

If we do that, then we could add an option to render terrain colors.

**Comment by WhistleWiz** [comment]
The vanilla one is not just a heightmap, it includes the wheat fields in the middle for example. Making a single screenshot with a camera would probably be the best way to go about it, then doing a bit of processing to the image to add an overlay of the borders or grid like the vanilla map.

**Comment by Insprill** [comment]
Ah, yeah, I was misremembering. Just having a color image would be good. Adding the grid coordinates and even ripped paper at the edges should be quite easy as well.
